interpreter/k8s/src/main/scala/io/buoyant/interpreter/k8s/IstioInterpreter.scala

```diff
@@ -0,0 +1,58 @@
+package io.buoyant.interpreter.k8s
+
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.PathMatcher
+import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.util.Activity
+import io.buoyant.namer.{ConfiguredDtabNamer, RewritingNamer}
+
+object IstioInterpreter {
+  private val noLabelsPfx = "/#/no-labels"
+  private val istioPfx = "/#/io.l5d.k8s.istio"
+  private val defaultRouteDtab = Dtab.read(s"""
+    |/svc/dest => /$$/inet ;
+    |/destPath/local/cluster/svc => $noLabelsPfx ;
+    |/svc/dest => /$$/io.buoyant.http.domainToPathPfx/destPath ;
+  """.stripMargin)
+
+  /* A typical delegation for the default route
+
+   /svc/dest/foo.default.svc.cluster.local/http
+   /$/io.buoyant.domainToPathPfx/destPath/foo.default.svc.cluster.local/http
+   /destPath/local/cluster/svc/default/foo/http
+   /#/no-labels/default/foo/http
+   /#/io.l5d.k8s.istio/default/foo/::/http
+
+   A typical delegation for a matching route
+   /svc/route/my-cool-route/http
+   /#/io.l5d.k8s.istio/default/reviews/version:v1/http
+  */
+
+  def apply(routeManager: Unit, istioNamer: Namer): NameInterpreter = {
+
+    // val routes = routerManager.routes
+    // TODO: Get routes from the route manager
+    val routes: Activity[IndexedSeq[Unit]] = Activity.value(IndexedSeq(()))
+
+    val routesDtab = routes.map { routeTable =>
+      val dentries = routeTable.map { _ =>
+        // TODO: map route into dentry
+        val route = "reviews-default"
+        val ns = "default"
```

Comment:
heads up that the route-rules from the api server don't include a namespace field, so we need to derive that from the fully qualified domain name or something